backend/kesaseteli/applications/jobs/daily/tests/test_clean_youth_applications.py

```diff
@@ -0,0 +1,132 @@
+from dateutil.relativedelta import relativedelta
+from django.utils import timezone
+
+from applications.jobs.daily import (
+    clean_inactive_youth_applications,
+    clean_old_youth_applications,
+    clean_rejected_youth_applications,
+)
+from applications.models import YouthApplication
+from common.tests.factories import (
+    AcceptedYouthApplicationFactory,
+    ActiveYouthApplicationFactory,
+    InactiveYouthApplicationFactory,
+    RejectableYouthApplicationFactory,
+    RejectedYouthApplicationFactory,
+    YouthApplicationFactory,
+)
+
+
+def setCreationDate(applications, created_at):
+    for application in applications:
+        application.created_at = created_at
+        application.save()
+    return applications
+
+
+@pytest.mark.django_db
+def test_clean_inactive_youth_applications_older_than_week():
+    date_week_ago = timezone.now() - relativedelta(weeks=1)
+    date_less_than_week_ago = timezone.now() - relativedelta(days=6)
+
+    # Create different applications
+    inactive_applications_created_week_ago = setCreationDate(
+        InactiveYouthApplicationFactory.create_batch(size=5),
+        created_at=date_week_ago,
+    )
+
+    inactive_applications_created_less_than_week_ago = setCreationDate(
+        InactiveYouthApplicationFactory.create_batch(size=5),
+        created_at=date_less_than_week_ago,
+    )
+    active_applications_created_week_ago = setCreationDate(
+        ActiveYouthApplicationFactory.create_batch(size=5),
+        created_at=date_week_ago,
+    )
+
+    # Run the batch job
+    clean_inactive_youth_applications.Job().execute()
+
+    # Inactive applications should have been cleaned
+    for application in inactive_applications_created_week_ago:
+        assert not YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Newer inactive applications should haven't been cleaned
+    for application in inactive_applications_created_less_than_week_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Active applications should haven't been cleaned
+    for application in active_applications_created_week_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+
+@pytest.mark.django_db
+def test_clean_rejected_youth_applications_older_than_year():
+    date_year_ago = timezone.now() - relativedelta(year=1)
+    date_less_than_year_ago = timezone.now() - relativedelta(years=1, days=-1)
+
+    # Create different applications
+    rejected_applications_created_year_ago = setCreationDate(
+        RejectedYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    rejected_applications_created_less_than_year_ago = setCreationDate(
+        RejectedYouthApplicationFactory.create_batch(size=5),
+        created_at=date_less_than_year_ago,
+    )
+
+    unhandled_applications_created_year_ago = setCreationDate(
+        RejectableYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
```

Comment:
You could use the UnhandledYouthApplicationFactory here. The only difference is the inclusion of YouthApplicationStatus.SUBMITTED i.e. the inactive applications in the set.

Reply:
fixed